pkg/service/service.go

```diff
@@ -1337,8 +1337,14 @@ func (s *Service) restoreServicesLocked() error {
 			restoredFromDatapath: true,
 		}
 
+		backendSet := map[string]struct{}{}
 		for j, backend := range svc.Backends {
 			hash := backend.L3n4Addr.Hash()
+
+			// skip when same port and ip address exist.
+			if _, found := backendSet[hash]; found {
+				continue
+			}
```

Comment:
Maybe I'm missing something, but who actually inserts into backendSet?

Please also add a unit test for this issue so we can assert that future changes don't break it.

Reply:
@borkmann 
I added inserts into backednSet.
and I will add a unit test for this issue.

Reply:
I added unit test.